From 6ebed2fa456aee6a35fb2146e871e9f5248e16a9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:04:10 -0400
Subject: [kernel] perf: Fix prototype of find_pmu_context()

Message-id: <1410181752-25631-6-git-send-email-jolsa@redhat.com>
Patchwork-id: 91686
O-Subject: [PATCH RHEL7.1 BZ1134356 005/307] perf: Fix prototype of find_pmu_context()
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 9e3170411ed171a126f4dca1672012a33efe59e5
Author: Mark Rutland <mark.rutland@arm.com>
Date: Mon Feb 10 17:44:18 2014 +0000

description
===========
For some reason find_pmu_context() is defined as returning void * rather
than a __percpu struct perf_cpu_context *. As all the requisite types are
defined in advance there's no reason to keep it that way.

This patch modifies the prototype of pmu_find_context to return a
__percpu struct perf_cpu_context *.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/events/core.c b/kernel/events/core.c
index a36f914..8217707 100644
--- a/kernel/events/core.c
+++ b/kernel/events/core.c
@@ -6310,7 +6310,7 @@ static int perf_event_idx_default(struct perf_event *event)
  * Ensures all contexts with the same task_ctx_nr have the same
  * pmu_cpu_context too.
  */
-static void *find_pmu_context(int ctxn)
+static struct perf_cpu_context __percpu *find_pmu_context(int ctxn)
 {
  struct pmu *pmu;
 
-- 
1.7.1